stl/inc/filesystem

```diff
@@ -1377,8 +1377,24 @@ namespace filesystem {
 #endif // ^^^ !_HAS_CXX20 ^^^
 
         _NODISCARD_FRIEND path operator/(const path& _Left, const path& _Right) { // append a pair of paths together
-            path _Tmp = _Left;
+            path _Tmp;
+            const auto _Right_size  = _Right._Text.size();
+            const auto _Right_first = _Right._Text.data();
+            const auto _Right_last  = _Right_first + _Right_size;
+
+            // Preallocate memory for the most common case
+            if (!_Has_drive_letter_prefix(_Right_first, _Right_last) && !_Is_slash(*_Right_first)) {
+                // !has_root_name(_Right) && !has_root_directory(_Right)

```

Comment:
Avoid unexplained code in comments, it should be either removed or added completely, or, at least, commented with explanation

Reply:
The point of this comment was to show the equivalence to higher-level functions, `operator=/` has similar comments.

I'm open to rewording suggestions.

Reply:
I guess then you should just discard my comment, if there's a precedent nearby

Reply:
Maybe merge this with 1384 so it looks less like commented-out code? i.e., `// Handle the most common case (!has_root_name(_Right) && !has_root_directory(_Right))`.